tests/CMakeLists.txt

```diff
@@ -122,7 +122,7 @@ file(GLOB_RECURSE runtime_test_files
   CONFIGURE_DEPENDS
   runtime/*
 )
-list(REMOVE_ITEM runtime_test_files ${CMAKE_CURRENT_SOURCE_DIR}/runtime/engine/*)
+list(REMOVE_ITEM runtime_test_files runtime/engine/cmake_vars.py)
```

Comment:
Just checking: this will trigger a copies to binary dir if engine files are changed, right? Which is good, meaning prev code was buggy?

Reply:
Correct that most engine files are now only copied if changed, but this is just a performance improvement and not a bug fix.

We previously took a glob on "runtime/engine/*" lower down on line 141 and called `configure_file()` for each of those, which was fine, but unnecessarily slow.

The bug was that this line (125) wasn't actually performing a glob - it was operating on the literal filename "runtime/engine/*". This mean that runtime engine files weren't removed from the list as expected and so were copied into the build directory, overwriting the versions that were placed there by `configure_file()`.

Ensuring that cmake_vars.py is properly removed from the list here and doesn't ever get copied was the bug fix.

Reply:
Ah, got it. Thanks for explaining